### PR TITLE
Feature/production monitoring 2

### PR DESCRIPTION
**Pull request by dblane-digicatapult**
# Pull Request

## Checklist
- [X] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [X] Feature
- [X] Breaking Change (fix or feature that would cause existing functionality to change)

## Linked tickets

https://digicatapult.atlassian.net/browse/SQNC-9

## High level description

PR 1 of 2
Implements the kube-prometheus-stack, CRDs and a nginx-ingress-controller to host grafana from.

## Detailed description

This implements prometheus and grafana using the kube-prometheus-stack.  It also implements CRDs for using prometheus.  An nginx-ingress-controller so that Grafana can be accessed externally.  The grafana instance is protected with an admin password.

## Describe alternatives you've considered

N/A

## Operational impact

This will install CRDs that will have to be manually removed from the cluster in the event that this PR is rolled back.

## Additional context

A future PR will add all of the service/pod watchers to individual services for Prometheus to discover/use.

### CODE REVIEW

File: clusters/l3-sqnc/monitoring/kube-prometheus-stack/values.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    selfMonitor: true
+    namespace: monitoring
+  config:
+    receivers:
+      - name: "null"
+      # - 
+      # - name: slack
+      #   webhook_configs:
+      #     - send_resolved: true
+      #       url: <secret>
```
Comment (mattdean-digicatapult) [inline clusters/l3-sqnc/monitoring/kube-prometheus-stack/values.yaml:13]: slack comes later?

File: clusters/l3-sqnc/monitoring/kube-prometheus-stack/values.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    podMonitorSelector:
+      matchLabels:
+        app.kubernetes.io/component: monitoring
+grafana:
+  defaultDashboardsEnabled: false
+  serviceMonitor:
+    enabled: true
+    namespace: monitoring
+  ingress:
+    enabled: true
```
Comment (mattdean-digicatapult) [inline clusters/l3-sqnc/monitoring/kube-prometheus-stack/values.yaml:56]: At the moment I assume grafana will be accessed using the admin password (which I presume you have somewhere?). I know the plan is to link this to our Okta SSO, is it worth disabling the ingress for now until we have that setup?

File: clusters/l3-sqnc/monitoring/kube-prometheus-stack/values.yaml (hunk 1 above)
Comment (dblane-digicatapult) [reply clusters/l3-sqnc/monitoring/kube-prometheus-stack/values.yaml:13]: That is correct.

File: clusters/l3-sqnc/monitoring/kube-prometheus-stack/values.yaml (hunk 2 above)
Comment (dblane-digicatapult) [reply clusters/l3-sqnc/monitoring/kube-prometheus-stack/values.yaml:56]: Grafana will be accessed via the admin password (In the secret we added in this PR).  It is fine to have the ingress enabled.

Comment (mattdean-digicatapult) [reply clusters/l3-sqnc/monitoring/kube-prometheus-stack/values.yaml:56]: Can I understand why you think that? I get it's secured in some way, but password sharing for operational activities is definitely a no no.

I'm only suggesting this not be on the public facing internet until we have auth properly in place

Comment (dblane-digicatapult) [reply clusters/l3-sqnc/monitoring/kube-prometheus-stack/values.yaml:56]: Only a limited subsection of the development team are capable of retrieving this secret, the same subset are the only ones capable of port-forwarding to the grafana service itself.

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**